JDBC.md

```diff
@@ -0,0 +1,684 @@
+LENGTH(string, CHARACTERS)                   | LENGTH(TRIM(TRAILING FROM string))         | Last optional parameters is not supported
+LTRIM(string)                                | TRIM(LEADING FROM string)                  |
+REPLACE(string1, string2, string3)           | REPLACE(string1, string2, string3)         |
+RIGHT(string, count)                         | SUBSTR(string, LENGTH(string) - count + 1) | Can produce too long native expression because of the string is used twice
+RTRIM(string)                                | TRIM(TRAILING FROM string)                 |
+SOUNDEX(string)                              | SOUNDEX(string)                            |
+SUBSTRING(string, start, length, CHARACTERS) | SUBSTR(string, start, length)              | Last optional parameters is not supported
+UCASE(string)                                | UPPER(string)                              |
+
+**Supported system scalar functions:**
+
+JDBC escape                      | Native                           | Comment
+-------------------------------- | -------------------------------- | ----------------------------------------------------------------
+DATABASE()                       | 'DEFAULT'                        | Tarantool does not support databases. Driver always replaces it by 'DEFAULT'
+IFNULL(expression1, expression2) | IFNULL(expression1, expression2) |
+USER()                           | <string literal>                 | Driver replaces the function to the current user name
+
+Let's take a look at a few examples of using scalar functions:
+
+```java
+Statement statement = connection.createStatement();
+// SELECT 'DEFAULT'
+ResultSet resultSet = statement.executeQuery("SELECT {fn database()}");
+```
+
+```java
+Statement statement = connection.createStatement();
+// SELECT UPPER('usa')
+ResultSet resultSet = statement.executeQuery("SELECT {fn ucase('usa')}");
+```
+
+```java
+Statement statement = connection.createStatement();
+// SELECT 2 * 3.141592653589793 * 3.141592653589793 / ABS(RANDOM() - ROUND(3.141592653589793, 4))
+ResultSet resultSet = statement.executeQuery(
+    "SELECT 2 * {fn pi()} * {fn pi()} / {fn abs({fn rand(252)} - {fn round({fn pi()}, 4)})}"
+);
+```
+
+#### Outer joins
+
+The escape syntax for an outer join is: `{oj <outer-join>}` where `<outer-join>` has the form:
+` table {LEFT|RIGHT|FULL} OUTER JOIN {table | <outer-join>} ON search-condition`
+
+The following SELECT statement uses the escape syntax for an outer join:
+
+```java
+Statement statement = connection.createStatement();
+// SELECT * FROM employee e LEFT OUTER JOIN department d ON e.dept_id = d.id
+ResultSet resultSet = statement.executeQuery(
+    "SELECT * FROM {oj employee e LEFT OUTER JOIN department d ON e.dept_id = d.id}"
+);
+```
+
+**Note**: even though the driver allows to use any available SQL join operators the Tarantool itself supports only the
+following variants: [NATURAL] LEFT [OUTER] JOIN, [NATURAL] INNER JOIN, or CROSS JOIN.
```

Comment:
I would explain a bit which types of joins are supported in tarantool, then would suggest to use only the subset that is supported by tarantool and the standard both within escape syntax. I guess this way we'll make clear what syntax can be used with tarantool and where it is appropriate to use escape syntax.

My wording:

> Tarantool supports the following JOIN clauses:
> 
> * Inner join: `table [NATURAL] [INNER] JOIN {table | joined-table} [<on-or-using>]`.
> * Left outer join: `table [NATURAL] LEFT [OUTER] JOIN {table | joined-table} [<on-or-using>]`.
> * Cross join: `table CROSS JOIN {table | joined-table}`.
> 
> ```
> <on-or-using> ::= ON search-condition | USING(columns-list)
> ```
> 
> `NATURAL`, `ON` and `USING` are mutually exclusive.
> 
> The driver does not perform full parsing of an escaped SQL clause, so it technically allows any JOIN inside `{oj <...>}` block. However it is recommended to use escape syntax only for outer joins, otherwise a request will not be portable and so using of escape syntax becomes pointless.
> 
> Since Tarantool does not support RIGHT and FULL outer joins, the following subset of clauses is both portable and supported:
> 
> ```
> <outer-join> ::= table LEFT OUTER JOIN {table | <outer-join>} ON search-condition
> ```